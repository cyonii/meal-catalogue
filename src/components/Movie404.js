import PropTypes from 'prop-types';
import Col from 'react-bootstrap/Col';

const Movies404 = ({ text }) => (
  <Col className="text-center mt-5">
    <h1 className="fs-3">
      <span className="text-danger fw-bold">404</span>
      <span>{ `${text} | not found` }</span>
    </h1>
  </Col>
);

Movies404.propTypes = {
  text: PropTypes.string,
};

Movies404.defaultProps = {
  text: '',
};

export default Movies404;
