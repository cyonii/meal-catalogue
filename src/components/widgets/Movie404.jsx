import PropTypes from "prop-types";

const Movies404 = ({ text }) => (
  <div
    className="h-[calc(100vh-64px)] flex justify-center items-center text-center py-10"
    data-testid="movie-404"
  >
    <h1 className="text-4xl dark:text-slate-50">
      <span className="text-red-500 font-bold">404</span>
      <span>{`${text} | not found`}</span>
    </h1>
  </div>
);

Movies404.propTypes = {
  text: PropTypes.string,
};

Movies404.defaultProps = {
  text: "",
};

export default Movies404;
