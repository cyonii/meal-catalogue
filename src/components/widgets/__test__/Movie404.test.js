import { render, screen, cleanup } from '@testing-library/react';
import Movie404 from '../Movie404';

describe('Movie404', () => {
  afterEach(() => cleanup());

  it('should render correctly', async () => {
    render(<Movie404 />);

    expect(screen.getByTestId('movie-404')).toBeInTheDocument();
    expect(screen.getByText('404')).toBeInTheDocument();
    expect(screen.getByText('404')).toHaveClass('text-danger');
    expect(screen.getByText('| not found')).toBeInTheDocument();
  });

  it('should render with given text', () => {
    render(<Movie404 text="Movies" />);

    expect(screen.getByText('Movies | not found')).toBeInTheDocument();
  });
});
