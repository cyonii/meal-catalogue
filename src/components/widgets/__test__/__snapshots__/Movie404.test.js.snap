// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie404 should render correctly 1`] = `
<div
  class="text-center mt-5 col"
  data-testid="movie-404"
>
  <h1
    class="fs-3"
  >
    <span
      class="text-danger fw-bold"
    >
      404
    </span>
    <span>
       | not found
    </span>
  </h1>
</div>
`;
